### ISSUE TITLE
Appimage support

### ISSUE THREAD

**Issue by drequivalent** [elementary/files#744]
This is more of a system-wide suggestion, but I decided to put it into Files repository because most of it is about Files.

# 1. Icon

Right now, AppImages show only as generic binaries:
![2018-11-10 21-41-35](https://user-images.githubusercontent.com/1463277/48304860-47c4b780-e532-11e8-9fe1-082e3046ac72.png)

But it doesn't have to be this way. Inside, it has a .desktop file:

```
[Desktop Entry]
Version=1.0
Type=Application
Name=qTox
GenericName=Tox client
Comment=Qt 5 based Tox instant messenger for secure communication
TryExec=qtox
Exec=qtox %u
Icon=qtox
Categories=Network;InstantMessaging;Chat;Telephony;VideoConference;
Terminal=false
MimeType=x-scheme-handler/tox;application/x-tox;
```

which points to an icon, which is also included in the appimage in all of its resolutions:

```
~/squashfs-root/local/share/icons/hicolor$ ls -lh
итого 60K
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 128x128
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 14x14
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 16x16
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 192x192
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 22x22
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 24x24
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 256x256
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 32x32
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 36x36
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 48x48
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 512x512
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 64x64
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 72x72
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 96x96
drwx------ 3 equivalent equivalent 4,0K ноя 10 21:42 scalable
```

So it is very much possible to extract a desktop file, extract an icon and show it in place of generic binary.

# 2. "Installing" an Appimage

The user should be able to "install" a verified AppImage simply by dragging it onto the Applications button on the Wingpanel and then into the Slingshot - very intuitive and helps keeping your apps in one place. Of course, effectively it will just move the Appimage file into their `.local/share/applications` folder or some other one, after which the app is available in the Slingshot.
Optionally, the user should be able to "install" an Appimage system-wide, which will require a SuDo, of course.
And, of course the "installation" will require a security hash check as described in issue #745

**Comment by jeremypw** [comment]
There two issues here, please split into to reports, thanks.

**Comment by peteruithoven** [comment]
Please refocus this issue on supporting AppImage icons. 
This issue: https://github.com/elementary/files/issues/745 doesn't really describe the AppImage install procedure you describe in this issue. Are you sure you don't want to open another issue on that?

**Comment by drequivalent** [comment]
Issue #745 describes the verification, that has to be done before installing the appimage. If the appimage is verified (already has been run and remembered), don't ask the user, just quietly move the file. If it wasn't ask if they are sure.

**Comment by cassidyjames** [comment]
Files is not an app launcher, so doing extra manual work to extract metadata from an app package and display it is outside the scope of this repository. If anything, it should be the thumbnailer's responsibility to extract that data and create a thumbnail, just like other filetypes.

**Comment by peteruithoven** [comment]
@cassidyjames you do perhaps have a link to the thumbnailer's repo? Somewhere to move this issue to?

Does elementary OS use tumbler?
https://docs.xfce.org/xfce/thunar/tumbler
```
$ apt show tumbler
Package: tumbler
Version: 0.2.1-0ubuntu1
Priority: optional
Section: universe/xfce
Origin: Ubuntu
Maintainer: Xubuntu Developers <xubuntu-devel@lists.ubuntu.com>
Original-Maintainer: Debian Xfce Maintainers <pkg-xfce-devel@lists.alioth.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 319 kB
Depends: tumbler-common (= 0.2.1-0ubuntu1), libc6 (>= 2.11), libcairo2 (>= 1.2.4), libfreetype6 (>= 2.2.1), libgdk-pixbuf2.0-0 (>= 2.22.0), libglib2.0-0 (>= 2.37.3), libgstreamer-plugins-base1.0-0 (>= 1.0.0), libgstreamer1.0-0 (>= 1.4.0), libjpeg8 (>= 8c), libpng16-16 (>= 1.6.2-1), libpoppler-glib8 (>= 0.18.0), libtumbler-1-0 (>= 0.1.27)
Suggests: tumbler-plugins-extra
Homepage: http://www.xfce.org/
Task: xubuntu-core, xubuntu-desktop, ubuntustudio-desktop-core, ubuntustudio-desktop
Supported: 3y
Download-Size: 70,7 kB
APT-Manual-Installed: no
APT-Sources: http://nl.archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
Description: D-Bus thumbnailing service
 Tumbler is a D-Bus service for applications to request thumbnails for various
 URI schemes and MIME types. It is an implementation of the thumbnail management
 D-Bus specification described on http://live.gnome.org/ThumbnailerSpec.
```

**Comment by cassidyjames** [comment]
@peteruithoven I'm not sure where the issue tracking repo is, but yes, we use tumbler afaik.
